From ae91c8c684c112a98193d9e5506e845c6a3386af Mon Sep 17 00:00:00 2001
From: Kyle Stein <kyle.stein@arm.com>
Date: Tue, 21 Jul 2020 14:52:21 -0500
Subject: [PATCH] Correctly handle empty string for proxy URI

Signed-off-by: Kyle Stein <kyle.stein@arm.com>
---
 cmd/edge-proxy/main.go | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/cmd/edge-proxy/main.go b/cmd/edge-proxy/main.go
index 6523516..e18e0fa 100644
--- a/cmd/edge-proxy/main.go
+++ b/cmd/edge-proxy/main.go
@@ -167,16 +167,17 @@ func main() {
 				fmt.Printf("Starting edge HTTP proxy (proxyAddr=%s, proxyURI=%s)\n", proxyAddr, proxyURI)
 
 				proxyForEdge := func(req *http.Request) (*url.URL, error) {
-					var proxy *url.URL
 
-					proxy, err := url.Parse(externalHTTPProxyURI)
-					if err != nil {
-						return nil, nil
-					} else {
-						return proxy, nil
+					if externalHTTPProxyURI != "" {
+						var proxy *url.URL
+						proxy, err := url.Parse(externalHTTPProxyURI)
+						if err == nil {
+							return proxy, nil
+						}
 					}
-				}
 
+					return nil, nil
+				}
 
 				server.RunEdgeHTTPProxyServer(childCtx, proxyAddr, forwardingAddresses(proxyURIParsed, forwardingAddressesMapParsed), caList, cert, proxyForEdge)
 
-- 
2.14.3

