diff --git a/edge-core/edge_server_customer_code.c b/edge-core/edge_server_customer_code.c
index 4e2b9a4..c17d520 100644
--- a/edge-core/edge_server_customer_code.c
+++ b/edge-core/edge_server_customer_code.c
@@ -33,6 +33,10 @@ bool edgeserver_execute_rfs_customer_code(edgeclient_request_context_t *request_
     } else {
         tr_info("edgeserver_execute_rfs_customer_code (local)");
     }
+
+    // Execute a script to do factory reset tasks, such as clearing customer logs.
+    int rc = system("edge-core-factory-reset");
+
     return true;
 }
 
