From c76608323f444001bc99571031d121bf1a80799d Mon Sep 17 00:00:00 2001
From: Nic Costa <nic.costa@gmail.com>
Date: Wed, 10 Mar 2021 15:28:28 -0600
Subject: [PATCH 12/15] add RAM total /3/0/3322 and RAM free /3/0/3321

---
 edge-client/gateway_stats.c | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/edge-client/gateway_stats.c b/edge-client/gateway_stats.c
index 5ca2e33..ad9bc21 100644
--- a/edge-client/gateway_stats.c
+++ b/edge-client/gateway_stats.c
@@ -32,6 +32,8 @@
 
 #define GATEWAY_STATS_CPU_TEMP_RES_ID 3303
 #define GATEWAY_STATS_CPU_PCT_RES_ID 3320
+#define GATEWAY_STATS_RAM_FREE_RES_ID 3321
+#define GATEWAY_STATS_RAM_TOTAL_RES_ID 3322
 
 /**
  * \struct cpu_info
@@ -226,6 +228,10 @@ void gsr_update_gateway_stats_resources(void *arg)
     // cpu usage
     gsr_set_resource_helper_float(GATEWAY_STATS_OBJ_ID, 0, GATEWAY_STATS_CPU_PCT_RES_ID, get_cpu());
 
+    // ram in bytes
+    const char cmd_ram_free[] = "awk '/^MemFree:/{ print $2*1024 }' /proc/meminfo";
+    gsr_set_resource_helper_int(GATEWAY_STATS_OBJ_ID, GATEWAY_STATS_RAM_FREE_RES_ID, int_exec(cmd_ram_free));
+
     return;
 }
 
@@ -234,6 +240,7 @@ void gsr_add_gateway_stats_resources()
 {
     int64_t int_default = 0;
     float float_default = 0;
+    int64_t int_actual;
 
     // cpu temp
     gsr_create_resource(GATEWAY_STATS_OBJ_ID,
@@ -257,5 +264,29 @@ void gsr_add_gateway_stats_resources()
                         sizeof(float_default),
                         NULL);
 
+    // ram total
+    const char cmd_ram_total[] = "awk '/^MemTotal:/{ print $2*1024 }' /proc/meminfo";
+    int_actual = int_exec(cmd_ram_total),
+    gsr_create_resource(GATEWAY_STATS_OBJ_ID,
+                        0,
+                        GATEWAY_STATS_RAM_TOTAL_RES_ID,
+                        "mem total",
+                        LWM2M_INTEGER,
+                        OPERATION_READ,
+                        (uint8_t *)&int_actual,
+                        sizeof(int_actual),
+                        NULL);
+
+    // ram free
+    gsr_create_resource(GATEWAY_STATS_OBJ_ID,
+                        0,
+                        GATEWAY_STATS_RAM_FREE_RES_ID,
+                        "mem free",
+                        LWM2M_INTEGER,
+                        OPERATION_READ,
+                        (uint8_t *)&int_default,
+                        sizeof(int_default),
+                        NULL);
+
     return;
 }
-- 
2.17.1

